tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -1054,6 +1056,37 @@ def test_group_experimental_warning() -> None:
         _ = TPESampler(multivariate=True, group=True)
 
 
+def run_tpe(k: int, sequence_dict: Dict[int, List[int]], hash_dict: Dict[int, int]) -> None:
```

Comment:
This function is only used in `test_group_deterministic_iteration`, so how about moving the definition inside `test_group_deterministic_iteration`? Then, this function does not need dictionaries as arguments.

Reply:
To use this function in multiprocessing, the function must be pickled.
Since local functions cannot be pickled, I put this on global scope.

https://stackoverflow.com/questions/8804830/python-multiprocessing-picklingerror-cant-pickle-type-function

I update the code and added some comments.

Reply:
Thanks you for the explanation. Sounds reasonable.